### PR TITLE
Early address selection

### PR DESCRIPTION
**Pull request by Atala**
- add address + date picker on restaurant list page
- use a react component on homepage with addresses suggestions

### CODE REVIEW

File: js/app/restaurant-list/RestaurantListFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    const inputProps = {
+      value: address,
+      onChange: (value) => { this.onAddressChange(value) },
+      onBlur: () => { this.onAddressBlur() },
+      placeholder: 'Entrez votre adresse'
+    }
+
+    if (!wantsNow) {
+      selectElement = <DatePicker
+                        setDeliveryDate={ this.props.onDatePickerChange }
```
Comment (alexsegura) [inline js/app/restaurant-list/RestaurantListFilter.jsx]: I have renamed it to `onChange` (naming things is hard haha)

File: js/app/restaurant-list/RestaurantListFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                        availabilities={ this.props.availabilities }
+                        deliveryDate={ this.props.initialDate }
```
Comment (alexsegura) [inline js/app/restaurant-list/RestaurantListFilter.jsx]: I have renamed it to `value` (same comment as above) 😄

**Comment by alexsegura** [comment]
Looks good to me. 

Just fix 2 lines and I think we can :shipit:

File: assets/css/vars.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Colors
```
Comment (alexsegura) [inline assets/css/vars.scss]: Nice to have introduced a file for variables 👍

File: js/app/cart/DatePicker.jsx [hunk 4]
```diff
@@ -102,7 +103,7 @@ class DatePicker extends Component {
             {
               dates.map(date => (
                 <option key={ date } value={ date } >
-                  { moment(date).format('dddd DD MMM') }
+                  { moment(date).calendar(null, { sameDay: '[Aujourd\'hui]',  nextDay: '[Demain]', sameElse: 'dddd DD MMM'}) }
```
Comment (alexsegura) [inline js/app/cart/DatePicker.jsx]: Can we use `window.__i18n`?

**Comment by Atala** [comment]
Note: at the end, the ideal would be to have the same `DatePicker` in coopcycle-web and in coopcycle-js.

**Comment by alexsegura** [comment]
@Atala yep, but on `coopcycle-js` there is Redux, and I had to adapt the code

File: js/app/cart/DatePicker.jsx (hunk 4 above)
Comment (Atala) [reply js/app/cart/DatePicker.jsx]: I think we should use a pure-JS solution (like [this](https://github.com/martinandert/counterpart) (didnt look into it) to translates our React component, `window.__i18n` is nice but don't scale.
I opened #99 to discuss this.

**Comment by Atala** [comment]
Merged !

**Pull request closed without merging**